Mirror/Runtime/NetworkTransformBase.cs

```diff
@@ -0,0 +1,545 @@
+        static float ScaleUShortToFloat(ushort value, ushort minValue, ushort maxValue, float minTarget, float maxTarget)
+        {
+            // note: C# ushort - ushort => int, hence so many casts
+            float targetRange = maxTarget - minTarget;
+            ushort valueRange = (ushort)(maxValue - minValue);
+            ushort valueRelative = (ushort)(value - minValue);
+            return minTarget + (float)((float)valueRelative/(float)valueRange * targetRange);
+        }
+
+        // ScaleByteToFloat(  0, byte.MinValue, byte.MaxValue, -1, 1) => -1
+        // ScaleByteToFloat(127, byte.MinValue, byte.MaxValue, -1, 1) => -0.003921569
+        // ScaleByteToFloat(191, byte.MinValue, byte.MaxValue, -1, 1) => 0.4980392
+        // ScaleByteToFloat(255, byte.MinValue, byte.MaxValue, -1, 1) => 1
+        static float ScaleByteToFloat(byte value, byte minValue, byte maxValue, float minTarget, float maxTarget)
+        {
+            // note: C# byte - byte => int, hence so many casts
+            float targetRange = maxTarget - minTarget;
+            byte valueRange = (byte)(maxValue - minValue);
+            byte valueRelative = (byte)(value - minValue);
+            return minTarget + ((float)valueRelative/(float)valueRange * targetRange);
+        }
+
+        // eulerAngles have 3 floats, putting them into 2 bytes of [x,y],[z,0]
+        // would be a waste. instead we compress into 5 bits each => 15 bits.
+        // so a ushort.
+        static ushort PackThreeFloatsIntoUShort(float u, float v, float w, float minValue, float maxValue)
+        {
+            // 5 bits max value = 1+2+4+8+16 = 31 = 0x1F
+            byte lower = ScaleFloatToByte(u, minValue, maxValue, 0x00, 0x1F);
+            byte middle = ScaleFloatToByte(v, minValue, maxValue, 0x00, 0x1F);
+            byte upper = ScaleFloatToByte(w, minValue, maxValue, 0x00, 0x1F);
+            ushort combined = (ushort)(upper << 10 | middle << 5 | lower);
+            return combined;
+        }
+
+        // see PackThreeFloatsIntoUShort for explanation
+        static float[] UnpackUShortIntoThreeFloats(ushort combined, float minTarget, float maxTarget)
+        {
+            byte lower = (byte)(combined & 0x1F);
+            byte middle = (byte)((combined >> 5) & 0x1F);
+            byte upper = (byte)(combined >> 10); // nothing on the left, no & needed
+
+            // note: we have to use 4 bits per float, so between 0x00 and 0x0F
+            float u = ScaleByteToFloat(lower, 0x00, 0x1F, minTarget, maxTarget);
+            float v = ScaleByteToFloat(middle, 0x00, 0x1F, minTarget, maxTarget);
+            float w = ScaleByteToFloat(upper, 0x00, 0x1F, minTarget, maxTarget);
+            return new float[]{u, v, w};
+        }
+
+        // serialization is needed by OnSerialize and by manual sending from authority
+        static bool SerializeIntoWriter(NetworkWriter writer, bool initialState, Vector3 position, Quaternion rotation, Compression compressRotation)
```

Comment:
remove unused parameter initialState

Reply:
The counterpart of this method is DeserializeFromReader,   it isn't static, and just uses the fields instead of receiving the parameters. This method should not be static either,  they should be mirror images of each other.

This comes into play if you add DataPoints as a list